providers/pool.rb

```diff
@@ -135,7 +135,11 @@ def configure
     doc = Document.new(xml)
 
     # root items
-    is_new_managed_runtime_version = new_value?(doc.root, 'APPPOOL/@RuntimeVersion', "v#{new_resource.runtime_version}")
+    if new_resource.no_managed_code
+      is_new_managed_runtime_version = new_value?(doc.root, 'APPPOOL/@RuntimeVersion', '')
+    else
+      is_new_managed_runtime_version = new_value?(doc.root, 'APPPOOL/@RuntimeVersion', "v#{new_resource.runtime_version}")
```

Comment:
The thing is that is_new_managed_runtime_version will be currently true even when there is no change (because empty RuntimeVersion means No Managed Code - but you are comparing it with string 'v', therefore the mismatch)